cw2dmk.c

```diff
@@ -193,7 +193,12 @@ msg(int level, const char *fmt, ...)
       !(level == OUT_RAW && out_file_level != OUT_RAW) &&
       !(level == OUT_HEX && out_file_level == OUT_RAW)) {
     va_start(args, fmt);
-    vfprintf(out_file, fmt, args);
+    if (vfprintf(out_file, fmt, args) < 0) {
+      fflush(stdout);
+      fprintf(stderr, "cw2dmk: Error writing to log file\n");
```

Comment:
Good change. As a style point, instead of `fprintf(stderr, "cw2dmk: ...\n"); perror("cw2dmk");`, I would write
`fprintf(stderr, "cw2dmk: ...: %s\n", strerror(errno));`

Reply:
Interesting!  Because that exactly how I had it originally coded and how I prefer to write it.  But then after reviewing the code base, I saw that perror() had been primarily used in cw2dmk.c and dmk2cw.c for fatal error reporting, even has recently as last month with 648929d6, so I rewrote my code to use perror() following the chameleon coding standard.

Should we have an shift where going forward avoiding perror() and using strerror() for reporting?

Reply:
I'm not so much against using perror at all, just against having both an fprintf(stderr) and a perror() for the same error. I see I did that in two places in dmk2cw.c, surprisingly, but today I'd consider that a bug.

Generally I tend to use perror in quickie programs, and then switch to fprintf later when cleaning up and improving error messages. Also I've gotten more careful in my coding style since I first wrote the Catweasel programs! Anyway, I think eventually getting rid of the remaining perror calls in favor of a nicer message would be a good cleanup.